File: KeeperFINAL/Models/Keep.cs
namespace KeeperFINAL.Models;
using KeeperFINAL.Interfaces;
using System.ComponentModel.DataAnnotations;

public class Keep : IHasCreator
{
  public int Id { get; set; }
  [Required]
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public int Kept { get; set; }
  public int Views { get; set; } = 0;


  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}

File: KeeperFINAL.client/src/models/Keep.js
import { Account } from "./Account.js"
export class Keep {
  constructor(data) {
    this.id = data.id
    this.creatorId = data.creatorId
    this.creator = data.creator
    // this.profile = data.profile
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.keptCount = data.kept
    this.viewCount = data.views
    this.vaultId = data.vaultId
    this.vaultKeepId = data.vaultKeepId
  }
}